src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5128,14 +5128,20 @@ private ScanTypeArgumentListKind ScanTypeArgumentList(NameOptions options)
                 case SyntaxKind.LessThanToken:           // e.g. `e is A<B> < C`
                 case SyntaxKind.LessThanEqualsToken:     // e.g. `e is A<B> <= C`
                 case SyntaxKind.GreaterThanEqualsToken:  // e.g. `e is A<B> >= C`
-                case SyntaxKind.GreaterThanToken when this.PeekToken(1).Kind != SyntaxKind.GreaterThanToken: // not >>
-                                                         // e.g. `e is A<B> > C`
                 case SyntaxKind.IsKeyword:               // e.g. `e is A<B> is bool`
                 case SyntaxKind.AsKeyword:               // e.g. `e is A<B> as bool`
-                    // These tokens are added to 7.5.4.2 Grammar Ambiguities in C#7 for a possible type following the `is` keyword
-                    return ((options & NameOptions.AfterIs) != 0)
-                        ? ScanTypeArgumentListKind.DefiniteTypeArgumentList
-                        : ScanTypeArgumentListKind.PossibleTypeArgumentList;
+                    // These tokens are added to 7.5.4.2 Grammar Ambiguities in C#7
```

Comment:
So the new behavior for `case List<string>[]` is to fall through to the `default` case and turn into a `PossibleTypeArgumentList`? I'm not sure I understand how this fixes the problem.

My thought process is as follows:
    1. The `<>` in `case List<string>[]` is a type argument list.
    2. It appears the previous behavior would return `DefiniteTypeArgumentList`.
    3. Why would `PossibleTypeArgumentList` be better than `DefiniteTYpeArgumentList` in this context. Where does `DefiniteTypeArgumentList` go off the rails?

Reply:
This code returns `DefiniteTypeArgumentList` for the `List<string>[]` case, as before. `PossibleTypeArgumentList` is not better in any way, as the bug being fixed here demonstrates.

---
In reply to: [139503034](https://github.com/dotnet/roslyn/pull/22102#discussion_r139503034) [](ancestors = 139503034)

Reply:
I think you misunderstand the flow of control here. The new behavior for case `List<string>[]` is to return `DefiniteTypeArgumentList`.

---
In reply to: [140905018](https://github.com/dotnet/roslyn/pull/22102#discussion_r140905018) [](ancestors = 140905018,139503034)

Reply:
AH, I see.